src/goto-instrument/code_contracts.cpp

```diff
@@ -1542,11 +1542,31 @@ goto_programt assigns_clauset::havoc_code(
   goto_programt havoc_statements;
   for(assigns_clause_targett *target : targets)
   {
+    //    if(target != NULL) goto x;
+    //      havoc_statements
+    // z: skip
+
+    // create the z label
+    goto_programt tmp_z;
+    goto_programt::targett z = tmp_z.add(goto_programt::make_skip(location));
+
+    // TODO: generalize this
+    // currently only supports pointers
```

Comment:
Could you turn this comment into a INVARIANT and open a GitHub issue to track this case?